healthcare/api-client/v1/dicom/dicom_stores_test.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2018 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+import sys
+import uuid
+
+import backoff
+from googleapiclient.errors import HttpError
+
+# Add datasets for bootstrapping datasets for testing
+sys.path.append(os.path.join(os.path.dirname(__file__), "..", "datasets"))  # noqa
+import datasets
+import dicom_stores
+
+cloud_region = "us-central1"
+project_id = os.environ["GOOGLE_CLOUD_PROJECT"]
+
+dataset_id = "test_dataset-{}".format(uuid.uuid4())
+dicom_store_id = "test_dicom_store_{}".format(uuid.uuid4())
+pubsub_topic = "test_pubsub_topic_{}".format(uuid.uuid4())
+
+RESOURCES = os.path.join(os.path.dirname(__file__), "resources")
+bucket = os.environ["CLOUD_STORAGE_BUCKET"]
+dcm_file_name = "dicom_00000001_000.dcm"
+content_uri = bucket + "/" + dcm_file_name
+dcm_file = os.path.join(RESOURCES, dcm_file_name)
+
+
+@pytest.fixture(scope="module")
+def test_dataset():
+    @backoff.on_exception(backoff.expo, HttpError, max_time=60)
+    def create():
+        try:
+            datasets.create_dataset(project_id, cloud_region, dataset_id)
+        except HttpError as err:
+            # We ignore 409 conflict here, because we know it's most
+            # likely the first request failed on the client side, but
+            # the creation suceeded on the server side.
+            if err.resp.status == 409:
+                print("Got exception {} while creating dataset".format(err.resp.status))
+            else:
+                raise
+
+    create()
+
+    yield
+
+    # Clean up
+    @backoff.on_exception(backoff.expo, HttpError, max_time=60)
+    def clean_up():
+        try:
+            datasets.delete_dataset(project_id, cloud_region, dataset_id)
+        except HttpError as err:
+            # The API returns 403 when the dataset doesn't exist.
+            if err.resp.status == 404 or err.resp.status == 403:
```

Comment:
What about 404? 

Can we update the comment to include why we are suppressing. Why do we want to surpress when it doesn't exist?

Reply:
It's a 403 when calling delete on a dataset that doesn't exist. Calling delete on a DICOM store that doesn't exist returns 404. Calling delete on a DICOM store that's in a dataset when the dataset doesn't exist returns a 403.

We can suppress it because if the dataset doesn't exist then we don't need to delete it.

Reply:
Can we add that to the comment :) (L85-87 is a good example)